### PR TITLE
Fix: tracking node files, affecting merges, update node files on release

### LINKED ISSUE #795: feat: add GitHub Actions release workflow and stop committing compiled dist/
## Summary

Implement **Path B** of the frontend build strategy discussed in #769.

Committing `qatrack/qatrack_core/static/dist/faults.js` (a compiled Vite build artifact) is the root cause of repeated merge conflicts experienced in #769 and the HTMX PR (#768). This issue tracks the work to eliminate that problem while keeping deployers self-contained via release archives.

---

### 1. `.gitignore`
Remove the `!qatrack/qatrack_core/static/dist/` override so the compiled bundle is properly gitignored going forward. Untrack the previously committed `faults.js`:

```bash
git rm --cached qatrack/qatrack_core/static/dist/faults.js
```

Vue source files (`qatrack/faults/static/faults/src/`) remain committed.

### 2. `.github/workflows/release.yml` (new file)
Triggered on `v*` tag pushes:
- Sets up Node 20, runs `npm ci && npm run build`
- Packages the app into a `.zip` (including the freshly built `faults.js`)
- Uploads the archive to a GitHub Release

Deployers who download a release archive need **zero** Node.js tooling.

### 3. `README.md`
Add a **"For contributors / development installs"** section explaining that git-clone developers must run `npm run build` after pulling frontend changes.

---

## Why

| | Current (commit dist/) | After this change |
|---|---|---|
| Merge conflicts from compiled JS | ✅ frequent | ❌ eliminated |
| Deployers need Node.js | ❌ no | ❌ no (release zip) |
| Git history cleanliness | ❌ noisy diffs | ✅ clean |
| Contributors need Node.js | implicitly yes | ✅ explicit, documented |

---

## References
- Discussion: #769 (comments)
- Related: HTMX PR #768
- Requested by @nsmela

### PR DESCRIPTION
**Pull request by nsmela**
# Pull Request Template

## Summary

Repo currently tracks node.js files that will change each time a dev generates them. This will cause issues with merging conflicts. Instead, a reelase CD is added to change these files only on releases. Additionally, a readme entry was added to notify others.

## Related Issues

Closes #795 which was raised with #769 and cuased by #768 

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that changes existing behaviour)
- [ ] Modernization / migration (part of the qatrackplus modernization effort)
- [ ] Documentation only
- [ ] Refactor / chore (no functional change)

## Target Branch

_Confirm which branch this targets (e.g. master)._

## AI Usage Disclosure

- [ ] No AI tools were used in this PR.
- [x] AI tools were used:
  - **Tools / where used:** Clause to generate based on direction from coderabbitai
  - [x] I've reviewed every AI-generated line as if I wrote it myself, and confirm: no PHI shared, no license-incompatible content (Apache-2.0).
---

### Coder Tasks

- [x] Rebased / merged `Dev` branch; no merge conflicts.
- [x] `uv run pre-commit run --all-files` passes (ruff lint, ruff-format, django-upgrade).
  - `uv run ruff check . --fix` to help resolve linting errors 
- [x] Django migrations generated and committed if models changed.
- [x] `uv run python -Wd manage.py check` reviewed for new deprecation warnings.
- [x] `uv run pytest` full test suite passes.
  - test fails with selenium, this is a pre-existing known issue
  - `uv run pytest -x -m "not selenium"` passes
- [x] Docker builds succeed if touched.
- [x] Docs updated under `docs/` if behaviour, settings, or APIs changed.
- [x] `release_notes.md` updated if user-facing.
- [x] No secrets, debug code, or stray `print`/`console.log` left in.

### Review Code

- [ ] Reviewer verifies the target branch.
- [ ] For non-trivial changes: pulled the branch, ran it locally, and confirmed it works as described.
- [ ] CI is green; test suite and `pre-commit` pass.
- [ ] Reads through all changed files for anything weird or unintended.
- [ ] Logic is correct; edge cases and error handling considered.
- [ ] Migrations are sane and reversible where possible.
- [ ] Changes are consistent with the modernization direction (no reintroduction
      of patterns being migrated away from, e.g. flake8/yapf-era conventions).
- [ ] No security concerns (input validation, permissions, injection, XSS).
- [ ] Comments are sufficient and not overkill.
- [ ] Typos, spellcheck, professional.
- [ ] Reviewer sends back review.

### Edits

- [ ] Coder fixes any issues.
- [ ] Reviewer approves.

> Maintainer review is required for breaking changes, new features, and migrations.
> For bug fixes, refactors, and docs-only changes, reviewer approval is sufficient to merge.

---

## Testing Notes

As required by steps above

## Screenshots

N/A

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - "v*"
```
Comment (nsmela) [inline .github/workflows/release.yml:6]: whenever we implement a tag that starts with "v", this action will kick in

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    steps:
+      - uses: actions/checkout@34e114876b0b11c390a56381ad16ebd13914f8d5
+        with:
+          persist-credentials: false
+
+      - name: Set up Node.js
+        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020
+        with:
+          node-version: "20"
```
Comment (nsmela) [inline .github/workflows/release.yml]: @coderabbitai is this the right node version? I recall our CI/CD warns of using this specific version

Comment (nsmela) [reply .github/workflows/release.yml]: @coderabbitai ok, commit this suggestion

**Review by crcrewso — CHANGES_REQUESTED** [review]
Looks good to me. Could you add some language to the Developer's guide near the top of the intro document?

**Comment by nsmela** [comment]
> Looks good to me. Could you add some language to the Developer's guide near the top of the intro document?

@coderabbitai please provide a plan to address this.

**Comment by nsmela** [comment]
@coderabbitai are we not using node 22?

**Comment by nsmela** [comment]
@coderabbitai yes, but first please review your suggestion for any other inaccuracies or missing info. Then plan.

**Comment by nsmela** [comment]
@coderabbitai provide the code in a comment for me to manually apply

**Comment by nsmela** [comment]
@crcrewso ready to review

**Review by nsmela — COMMENTED** [review]
squashed and rebased to latest Dev commit

**Review by crcrewso — APPROVED** [review]
Looks good to me!
